Sources/App/Controllers/PackageController+routes.swift

```diff
@@ -368,30 +368,25 @@ enum PackageController {
 
         // For repositories that have no README file at all.
         guard let readmeHtmlUrl = pkg.repository?.readmeHtmlUrl else {
-            return PackageReadme.View(model: .init(url: nil,
-                                                   repositoryOwner: nil,
-                                                   repositoryName: nil,
-                                                   defaultBranch: nil,
-                                                   readme: nil)).document()
+            return PackageReadme.View(model: .noReadme).document()
         }
 
         do {
             let readme = try await Current.fetchS3Readme(req.client, owner, repository)
+            guard let branch = pkg.repository?.defaultBranch else {
+                return PackageReadme.View(model: .cacheLookupFailed(url: readmeHtmlUrl)).document()
```

Comment:
I'm using `cacheLookupFailed` here, because it kinda is (we can't hook up the cache without the branch) - although this could never really happen. I still want to model the fields as non-optional in the db one day...